lib/content-services/src/lib/dialogs/dialog/dialog.component.html

```diff
@@ -32,26 +32,50 @@ <h2 class="adf-dialog-title">{{data.title}}</h2>
         [ngTemplateOutlet]="data.descriptionTemplate"
         [ngClass]="{ 'adf-description': data.description || data.descriptionTemplate }"
     >
-        <ng-container>{{data.description}}</ng-container>
+        <ng-container>{{ data.description }}</ng-container>
     </div>
 
-    <mat-dialog-content class="adf-dialog-content">
+    <mat-dialog-content *ngIf="data.contentTemplate || data.contentComponent || data.contentText" class="adf-dialog-content">
         <ng-container [ngTemplateOutlet]="data.contentTemplate"></ng-container>
+        <ng-container [ngComponentOutlet]="data.contentComponent"></ng-container>
+        <ng-container>{{ data.contentText }}</ng-container>
     </mat-dialog-content>
 
-    <mat-dialog-actions class="adf-dialog-actions" [ngClass]="{ 'adf-additional-actions': data.actionsTemplate }">
+    <mat-dialog-actions
+        class="adf-dialog-actions"
+        [ngClass]="{ 'adf-additional-actions': data.actionsTemplate || additionalActionButton1 || additionalActionButton2}"
+    >
         <div>
-            <ng-container [ngTemplateOutlet]="data.actionsTemplate"></ng-container>
+            <ng-container *ngIf="data.actionsTemplate && !(additionalActionButton1 || additionalActionButton2)" [ngTemplateOutlet]="data.actionsTemplate"></ng-container>
+            <ng-container *ngIf="(additionalActionButton1 || additionalActionButton2) && !data.actionsTemplate">
+                <button
+                    mat-button
+                    color="primary"
+                    class="{{ additionalActionButton1?.class }}"
+                    (click)="additionalActionButton1.onClick()"
```

Comment:
should be nullable as the button might not be defined
```suggestion
                    (click)="additionalActionButton1?.onClick()"
```

Reply:
good catch, thank you!